src/store-api/src/region_request.rs

```diff
@@ -740,6 +743,68 @@ impl AlterKind {
                     .iter()
                     .try_for_each(|name| Self::validate_column_to_drop(name, metadata))?;
             }
+            AlterKind::SyncColumns { column_metadatas } => {
+                let new_primary_keys = column_metadatas
+                    .iter()
+                    .filter(|c| c.semantic_type == SemanticType::Tag)
+                    .map(|c| (c.column_schema.name.as_str(), c.column_id))
+                    .collect::<HashMap<_, _>>();
+
+                let old_primary_keys = metadata
+                    .column_metadatas
+                    .iter()
+                    .filter(|c| c.semantic_type == SemanticType::Tag)
+                    .map(|c| (c.column_schema.name.as_str(), c.column_id));
```

Comment:
Primary keys may have a different order than the original column order.

We can create a table with columns (a, b, c) but primary key (c, a).